base/io.jl

```diff
@@ -647,11 +647,11 @@ Commit all currently buffered writes to the given stream.
 flush(io::IO) = nothing
 
 """
-    skipchars(io::IO, predicate; linecomment=nothing)
+    skipchars(io::IO, predicate; linecomment::Char)
 
-Skip forward in `io` until `predicate` returns `false`. If `linecomment`
-is defined, all characters after the `linecomment` character are ignored
-until the next line.
+Advance the stream `io` until before the first character for which `predicate`
```

Comment:
I felt that this (from the original docs) was actually more clear and more accurate than my original version. "Before the first character for which `predicate` returns `false`" means that the next character to be read will be false.

The way I see it, the stream cannot really be advanced to a character. Does this mean the next character the stream reads is the one its "at" or the one after? Instead the stream position is always between characters, meaning that the character in front of the stream is about to be read and the one after was just read (should we be reading through the stream one character after the other).

From this perspective "before the first character..." has to mean the next character to be read is the one that returns `false` and "Advance to character..." doesn't really make sense. That said, if there is some established convention for this kind of thing I happy to work with that.